sdk/core/core/CMakeLists.txt

```diff
@@ -62,7 +63,13 @@ else()
   target_compile_options(az_core PRIVATE -Wall -Wextra -pedantic  ${WARNINGS_AS_ERRORS_FLAG})
 endif()
 
-target_include_directories (az_core PUBLIC inc)
+target_include_directories (az_core PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/inc> $<INSTALL_INTERFACE:include>)
+
+# create interface for internal headers
+add_library(az_core_internal INTERFACE)
+target_include_directories(az_core_internal INTERFACE internal)
```

Comment:
because we want to allow seperate building this will need the `$<BUILD_INTERFACE:...>` and `$<INSTALL_INTERFACE:...>` stuff as well.

Reply:
I'm OK with accepting this PR without this, as we can add it in later with relative ease, and our install rules are WIP anyway.